src/main/java/org/codehaus/plexus/components/io/filemappers/SuffixFileMapper.java

```diff
@@ -0,0 +1,81 @@
+package org.codehaus.plexus.components.io.filemappers;
+
+/*
+ * Copyright 2007 The Codehaus Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import javax.annotation.Nonnull;
+
+import org.codehaus.plexus.util.StringUtils;
+
+/**
+ * A file mapper, which maps by adding a suffix.
+ */
+public class SuffixFileMapper extends AbstractFileMapper
+{
+    /**
+    * The suffix mappers role-hint: "suffix".
+    */
+    public static final String ROLE_HINT = "suffix";
+
+    private String suffix;
+
+    @Nonnull public String getMappedFileName( @Nonnull String name )
+    {
+        final String s = super.getMappedFileName( name ); // Check for null, etc.
+        return getMappedFileName( suffix, s );
+    }
+
+    /**
+     * Returns the suffix to add.
+     */
+    public String getSuffix()
+    {
+        return suffix;
+    }
+
+    /**
+     * Sets the suffix to add.
+     */
+    public void setSuffix( String suffix )
+    {
+        this.suffix = suffix;
+    }
+
+    /**
+     * Performs the mapping of a file name by adding a suffix.
+     */
+    public static String getMappedFileName( String suffix, String name )
```

Comment:
I guess you followed the `PrefixFileMapper` code, but the rest does not have `public static String getMappedFileName` and I think it would be best to be consistent with them as `PrefixFileMapper` is a kind of exception. If you don't have other concerns I think it's best to move it to `public String getMappedFileName( @Nonnull String name )`.

Reply:
Ok I have no problem with that
Indeed like you said I'm basically inspired of PrefixFileMapper because I didn't know plexus-io :(